server/services/applicationService.ts

```diff
@@ -63,6 +63,32 @@ export default class ApplicationService {
     return page
   }
 
+  getResponses(application: Application) {
+    if (application.data) {
+      return Object.fromEntries(
```

Comment:
this function is a little hard for me to understand 😭
A couple of ideas that might make it easier on the/my eyes : 
- We're retrieving a Page object in a couple of places in this service and throwing if we don't find it, I wonder if it would be worth extracting this to a util function so we don't have to have 
``` 
              if (!Page) {
                throw new UnknownPageError()
              }
```
duplicated?
- could application.data been undefined? if so the type should reflect this - otherwise we can remove the check on line 67
- I think some temp variables would better describe what's going on, if we saved the result of the function on line 73 as 'pageResponses'(?) or something I think it would be easier to work out what's going on

Reply:
I've had another crack at this 👍